app/controllers/admin/genres_controller.rb

```diff
@@ -12,8 +11,8 @@ def new
 
  def create
    @genre = Genre.create(genre_params)
-   if @genre.save
-     redirect_to admin_genres_path
+   if current_admin? && @genre.save
```

Comment:
`current_admin?` would have been implied with the `before_action` method you removed